csrc/megatron/fused_rotary_positional_embedding_cuda.cu

```diff
@@ -103,8 +102,159 @@ torch::Tensor bwd_cuda(const torch::Tensor &output_grads,
       dispatch_fused_rope_backward(
           s, b, h, d, d2, stride_s, stride_b, stride_h, stride_d, o_stride_s,
           o_stride_b, o_stride_h, o_stride_d,
-          output_grads.data_ptr<scalar_t_0>(), cos.data_ptr<scalar_t_0>(),
-          sin.data_ptr<scalar_t_0>(), input_grads.data_ptr<scalar_t_0>());)
+          output_grads.data_ptr<scalar_t_0>(), freqs.data_ptr<float>(),
+          input_grads.data_ptr<scalar_t_0>()););
+  return input_grads;
+}
+
+#define DISPATCH_FUSED_ROPE_TYPES(TYPE1, TYPE2, NAME, ...)                     \
```

Comment:
Because we don't need 9 combinations (float, fp16, bf16) x (float, fp16, bf16). We want dtype of the second operand to be either `fp32` or same as the first operand.